### PR TITLE
Feat: Update Regionalization Button and Regionalization Bar to show the postal code

### PR DESCRIPTION
**Pull request by saranicoly**
Applies patch from https://github.com/vtex-sites/base.store/pull/495

## What's the purpose of this pull request?

Update Regionalization Button and Regionalization Bar to show the last postal code saved, instead of receiving a `content` prop.

image

image

## How to test it?

1- Set a valid postal code in `RegionalizationInput` 
2- Press "enter" key
3- Push the close button.

## References

- [Base store's PR](https://github.com/vtex-sites/base.store/pull/495)
- [Postal Code Bar's original code](https://github.com/vtex-sites/base.store/pull/424)

## Checklist

- [x] CHANGELOG entry added

### CODE REVIEW

**Review by filipewl — APPROVED** [review]
LGTM!

This may be more of a design/UX question, but do you know if there is going to be a way for the user to remove the zip code via the UI? 

Once a zip code is set, as of now they can only replace it. Or, if they're technical, and want to remove it, they need to delete the entry in the IndexedDB via DevTools. 

I wonder if there's a real use case for removing the zip code to check the difference in prices, product availability, etc.

**Comment by igorbrasileiro** [comment]
> This may be more of a design/UX question, but do you know if there is going to be a way for the user to remove the zip code via the UI?
@filipewl, at other stores, like c4, this flow doesn't exist, you can only update to new one.

File: src/components/regionalization/RegionalizationBar/RegionalizationBar.tsx [hunk 1]
```diff
@@ -2,26 +2,26 @@ import type { HTMLAttributes } from 'react'
...
 
   return (
     <div data-fs-regionalization-bar className={classes} {...otherProps}>
       <Button onClick={() => setIsRegionalizationModalOpen(true)}>
         <Icon name="MapPin" width={24} height={24} />
-        {content ? (
+        {postalCode ? (
           <>
-            <span>{content}</span>
+            <span>{postalCode}</span>
```
Comment (igorbrasileiro) [inline src/components/regionalization/RegionalizationBar/RegionalizationBar.tsx:24]: Add key to this span and the other below that has "set your location".

File: src/components/regionalization/RegionalizationButton/RegionalizationButton.tsx [hunk 2]
```diff
@@ -24,9 +24,9 @@ export default function RegionalizationButton({
...
         onClick={() => setIsRegionalizationModalOpen(true)}
       >
-        {content ? (
+        {postalCode ? (
           <>
-            <span>{content}</span>
+            <span>{postalCode}</span>
           </>
         ) : (
           <span>Set your location</span>
```
Comment (igorbrasileiro) [inline src/components/regionalization/RegionalizationButton/RegionalizationButton.tsx:32]: Remove the fragment, ins't necessary. Also, add key on both span.

**Review by igorbrasileiro — COMMENTED** [review]
Left comments

File: src/components/regionalization/RegionalizationButton/RegionalizationButton.tsx (hunk 2 above)
Comment (filipewl) [reply src/components/regionalization/RegionalizationButton/RegionalizationButton.tsx:32]: Good call on the unneeded fragment! I wonder if merging the 2 `span`s into 1 like the following would also work?

```jsx
<span>{ postalCode ?? 'Set your location' }</span>
```

File: src/components/regionalization/RegionalizationBar/RegionalizationBar.tsx (hunk 1 above)
Comment (filipewl) [reply src/components/regionalization/RegionalizationBar/RegionalizationBar.tsx:24]: Could you explain why the key is needed here? I know it's normally used when rendering an array of items, but know very little of other use cases.

I found this other use case (https://kentcdodds.com/blog/understanding-reacts-key-prop), but not sure it's the same here.

File: src/components/regionalization/RegionalizationButton/RegionalizationButton.tsx (hunk 2 above)
Comment (saranicoly) [reply src/components/regionalization/RegionalizationButton/RegionalizationButton.tsx:32]: Thanks for the suggestion, @filipewl. It worked.

Comment (saranicoly) [reply src/components/regionalization/RegionalizationButton/RegionalizationButton.tsx:32]: @filipewl It worked, but I think it's best to keep the spans separated in order to have different keys. What do you think?

File: src/components/regionalization/RegionalizationBar/RegionalizationBar.tsx (hunk 1 above)
Comment (saranicoly) [reply src/components/regionalization/RegionalizationBar/RegionalizationBar.tsx]: Thanks for bringing this up ☺️

**Review by igorbrasileiro — APPROVED** [review]

File: src/components/regionalization/RegionalizationBar/RegionalizationBar.tsx (hunk 1 above)
Comment (igorbrasileiro) [reply src/components/regionalization/RegionalizationBar/RegionalizationBar.tsx]: Yeah, i was wrong asking to add keys, didn't make sense

**Pull request merged**